managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1494,71 +1503,67 @@ public void deleteFailed(ManagedLedgerException exception, Object ctx) {
         }
     }
 
-    @Override
-    public void asyncDelete(Position pos, final AsyncCallbacks.DeleteCallback callback, Object ctx) {
-        checkArgument(pos instanceof PositionImpl);
 
-        if (STATE_UPDATER.get(this) == State.Closed) {
+    @Override
+    public void asyncDelete(Iterable<Position> positions, AsyncCallbacks.DeleteCallback callback, Object ctx) {
+        if (state == State.Closed) {
```

Comment:
Leave it as STATE_UPDATER.get(this)

Reply:
There's no difference with `STATE_UPDATER.get(this)`. Reading a volatile variable gives the last value of it.